core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -138,20 +137,25 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
 
   override def preStart(): Unit = {
     super.preStart()
-    requests = Map.empty[TopicPartition, ActorRef]
     consumer = settings.createKafkaConsumer()
-    commitsInProgress = 0
   }
 
   override def postStop(): Unit = {
     pollTask.cancel()
+
+    // reply to outstanding requests is important if the actor is restarted
+    requests.foreach {
+      case (ref, req) =>
+        ref ! Messages(req.requestId, Iterator.empty)
+    }
+
     consumer.close()
     super.postStop()
   }
 
   def poll() = {
     //set partitions to fetch
-    val partitionsToFetch = requests.keys.toSet
+    val partitionsToFetch: Set[TopicPartition] = requests.values.flatMap(_.topics)(collection.breakOut)
```

Comment:
Then `breakOut` can't be used. For great performance we need to use other collections anyway. We need to get the logic right (and understandable) first, though.